syntax = "proto3";

package Imagine_MapReduce.Internal;
import "InternalType.proto";

// Mapper提供的Map函数 √

message MapTaskRequestMessage {
    Identity send_identity_ = 1;                // 发送者身份
    Identity recv_identity_ = 2;                // 接收者身份
    string file_name_ = 3;                      // 文件名
    uint32 split_size_ = 4;                     // split大小, 单位为B
    string listen_ip_ = 5;                      // master监听Ip
    string listen_port_ = 6;                    // master监听Port
}

message MapTaskResponseMessage {
    Identity send_identity_ = 1;                // 发送者身份
    Identity recv_identity_ = 2;                // 接收者身份
    Status status_ = 3;                         // 任务接收状态
}